R/partial-factor.R

```diff
@@ -50,7 +50,7 @@ new_partial_factor <- function(partial = factor(), learned = factor()) {
 vec_ptype_full.vctrs_partial_factor <- function(x, ...) {
   empty <- ""
 
-  levels <- map(x, levels)
+  levels <- map(unclass(x), levels)
```

Comment:
Why does this change?

Reply:
To avoid an infinite recursion.

Reply:
But why did it work before?

Reply:
Because `map()` didn't use vctrs operations and genericity.

The purrr compat file has been updated to use the vec_map() to get some internal testing and to make it easy to import unit tests from purrr.

Reply:
oops sorry, the infinite recursion was another problem. It doesn't make sense for ptype-full to be recursed into anyway.

The problem is that `map()` now assigns names, and partial factors and data frames don't support that:

```
test-partial-factor.R:5: error: has ok print method
`names<-.vctrs_partial_factor()` not supported.
```

This is a flaw in the partial types but as usual I'm just working around these types when they cause problems for now.

Reply:
We no longer assign `NULL` names to be a little more efficient. However this `unclass()` change is still needed because partial types inherit from `vctrs_sclr` which have an unsupported `names<-` method but still allow `names()`. Making the latter unsupported causes a bunch of other issues. It wouldn't solve the problem at hand anyway because now we'd have a vector type for which `names()` is an error, which is a big genericity flaw. I think we shouldn't worry about these types for now.